From dfc526596c890ef5877deea8a405c7562fc464da Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:30 -0400
Subject: [usb] xhci: Switch only Intel Lynx Point-LP ports to EHCI on shutdown

Message-id: <1410887463-20674-499-git-send-email-dzickus@redhat.com>
Patchwork-id: 95436
O-Subject: [RHEL7 PATCH 498/591] xhci: Switch only Intel Lynx Point-LP ports to EHCI on shutdown.
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 0a939993bff117d3657108ca13b011fc0378aedb
Author: Denis Turischev <denis.turischev@compulab.co.il>
Date:   Tue May 20 14:00:42 2014 +0300

    xhci: Switch only Intel Lynx Point-LP ports to EHCI on shutdown.

    Patch "xhci: Switch Intel Lynx Point ports to EHCI on shutdown."
    commit c09ec25d3684cad74d851c0f028a495999591279 is not fully correct

    It switches both Lynx Point and Lynx Point-LP ports to EHCI on shutdown.
    On some Lynx Point machines it causes spurious interrupt,
    which wake the system: bugzilla.kernel.org/show_bug.cgi?id=76291

    On Lynx Point-LP on the contrary switching ports to EHCI seems to be
    necessary to fix these spurious interrupts.

    Signed-off-by: Denis Turischev <denis@compulab.co.il>
    Reported-by: Wulf Richartz <wulf.richartz@gmail.com>
    Cc: Mathias Nyman <mathias.nyman@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index ffd119e..e20520f 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -134,7 +134,9 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
    */
   if (pdev->subsystem_vendor == PCI_VENDOR_ID_HP)
    xhci->quirks |= XHCI_SPURIOUS_WAKEUP;
-
+ }
+ if (pdev->vendor == PCI_VENDOR_ID_INTEL &&
+  pdev->device == PCI_DEVICE_ID_INTEL_LYNXPOINT_LP_XHCI) {
   xhci->quirks |= XHCI_SPURIOUS_REBOOT;
  }
  if (pdev->vendor == PCI_VENDOR_ID_ETRON &&
-- 
1.7.1